### PR TITLE
TRUNK-5970: Encounter to handle voided conditions.

### PR DESCRIPTION
**Pull request by mks-d**
That's the backport to 2.3.x.

### CODE REVIEW

**Review by ibacher — APPROVED** [review]

File: api/src/main/java/org/openmrs/Encounter.java [hunk 1]
```diff
@@ -450,18 +451,27 @@ public void setDiagnoses(Set<Diagnosis> diagnoses) {
...
-		return conditions;
+		return getConditions(false);
 	}
 
+	/**
+	 * Returns all conditions where 'Condition.encounterId = Encounter.encounterId'.
+	 *
+	 * @param includeVoided - Specifies whether or not to include voided conditions.
+	 * @return The set of conditions, or an empty set if there are no conditions to return.
+	 * @since 2.5.0
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/Encounter.java:468]: Shouldn't we also add a 2.3.3?

Comment (ibacher) [reply api/src/main/java/org/openmrs/Encounter.java:468]: Good point!

**Comment by dkayiwa** [comment]
@ibacher in case you just forgot to update the JIRA ticket. 😊

**Comment by ibacher** [comment]
Updated Jira and fixed the `@since` documentation to read `@since 2.3.3, 2.4.0, 2.5.0`.

**Pull request merged**